src/app/core/services/scheduling.service.ts

```diff
@@ -54,22 +54,18 @@ export class SchedulingService {
       if (schedule) {
         const startDate = this.setDateTimeToMidnight(date)
         const endDate = this.advanceRepeat(startDate, 'day', 1)
-        let tasks: Task[] = []
+        const tasks: Task[] = []
         for (let i = 0; i < schedule.length; i++) {
-          if (
-            schedule[i].timestamp < endDate.getTime() &&
-            schedule[i].timestamp > startDate.getTime()
-          ) {
+          if (schedule[i].timestamp > endDate.getTime()) break
+          if (schedule[i].timestamp > startDate.getTime())
```

Comment:
I've changed the schedule stored in storage to be sorted already instead of sorting it every time it retrieves the task. So this break makes it not check the rest of the schedule.

Reply:
Okay good idea. Thanks